app/src/main/java/org/gdg/frisbee/android/api/GroupDirectoryFactory.java

```diff
@@ -1,44 +1,44 @@
 package org.gdg.frisbee.android.api;
 
-import org.gdg.frisbee.android.BuildConfig;
+import org.gdg.frisbee.android.app.App;
 import org.gdg.frisbee.android.utils.Utils;
 
-import retrofit.RequestInterceptor;
-import retrofit.RestAdapter;
-import retrofit.converter.GsonConverter;
-
-/**
- * Created by <a href="mailto:marcusandreog@gmail.com">Marcus Gabilheri</a>
- *
- * @author Marcus Gabilheri
- * @version 1.0
- * @since 7/26/15.
- */
+import java.io.IOException;
+
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import retrofit2.GsonConverterFactory;
+import retrofit2.Retrofit;
+
 public final class GroupDirectoryFactory {
 
-    // This variable is meant to be changed as pleased while debugging
-    private static final RestAdapter.LogLevel LOG_LEVEL = RestAdapter.LogLevel.NONE;
     private static final String BASE_URL = "https://developers.google.com";
 
     private GroupDirectoryFactory() {
     }
 
-    private static RestAdapter provideRestAdapter() {
-        return new RestAdapter.Builder()
-                    .setEndpoint(BASE_URL)
-                    .setConverter(new GsonConverter(Utils.getGson()))
-                    .setLogLevel(BuildConfig.DEBUG ? LOG_LEVEL : RestAdapter.LogLevel.NONE)
-                    .setClient(OkClientFactory.provideClient())
-                    .setRequestInterceptor(new RequestInterceptor() {
-                        @Override
-                        public void intercept(RequestFacade request) {
-                            request.addHeader("User-Agent", "GDG-Frisbee/0.1 (Android)");
-                            request.addHeader("Referer", "https://developers.google.com/groups/directory/");
-                            request.addHeader("X-Requested-With", "XMLHttpRequest");
-                            request.addHeader("Cache-Control", "no-cache");
-                            request.addHeader("DNT", "1");
-                        }
-                    })
+    private static Retrofit provideRestAdapter() {
+        OkHttpClient.Builder client = App.getInstance().getOkHttpClient().newBuilder();
+        client.addInterceptor(new Interceptor() {
+            @Override
+            public Response intercept(Chain chain) throws IOException {
+
+                Request compressedRequest = chain.request().newBuilder()
+                        .header("User-Agent", "GDG-Frisbee/0.1 (Android)")
```

Comment:
This is the Client for `developers.google.com` which is going to be deprecated in the future. We are using only 2 API calls from here.